internal/backend/dispatcher/eventsworkflow.go

```diff
@@ -181,14 +181,36 @@ func (d *dispatcher) signalWorkflows(ctx context.Context, event sdktypes.Event)
 		return nil
 	}
 
-	signals, err := d.db.ListSignalsWaitingOnConnection(ctx, conn.ID(), event.Type())
+	signals, err := d.db.ListSignalsWaitingOnConnection(ctx, conn.ID())
 	if err != nil {
 		z.Error("could not fetch signals", zap.Error(err))
 		return err
 	}
 
 	z.Debug("found signal candidates", zap.Int("count", len(signals)))
 	for _, signal := range signals {
+		match, err := event.Matches(signal.Filter)
+		if err != nil {
+			z.Error("inavlid signal filter",
+				zap.Error(err),
+				zap.String("signal_id", signal.SignalID),
+				zap.String("filter", signal.Filter),
+				zap.String("event_id", event.ID().String()))
```

Comment:
you can put all of these in a `With` statement above, then use the logger without these.

Reply:
ok, added